File: DSA/PriyankaSingh_590016827/day52question2.c
#include <stdio.h>

int main() {
    int n, k;
    printf("Enter number of elements: ");
    scanf("%d", &n);

    int arr[n];
    printf("Enter %d elements:\n", n);
    for(int i = 0; i < n; i++)
        scanf("%d", &arr[i]);

    printf("Enter the difference value k: ");
    scanf("%d", &k);

    printf("Pairs with difference %d:\n", k);
    for(int i = 0; i < n; i++) {
        for(int j = 0; j < n; j++) {
            if(i != j && arr[i] - arr[j] == k)
                printf("(%d, %d)\n", arr[i], arr[j]);
        }
    }

    return 0;
}

File: DSA/VaniAgrawal_590015846/Day_52_Question_2.py
# Implement systematic pair finding using hash-based lookup or sorted traversal to identify all pairs meeting difference criteria comprehensively.

arr = list(map(int, input("Enter array elements separated by spaces: ").split()))
k = int(input("Enter difference k: "))

s = set(arr)
pairs = []

for num in arr:
    if (num + k) in s:
        pairs.append((num, num + k))

print("Output:", pairs)
